study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/TestBostonBackfillCompleted.java

```diff
@@ -0,0 +1,180 @@
+        }
+        if (statusChangeEvent.getPostDelaySeconds() == null || statusChangeEvent.getPostDelaySeconds() <= 0) {
+            throw new DDPException("Event does not have delaySeconds setting");
+        }
+
+        runBackfill(handle, studyDto, statusChangeEvent);
+    }
+
+    private void runBackfill(Handle handle, StudyDto studyDto, EventConfiguration event) {
+        var helper = handle.attach(SqlHelper.class);
+        long delayMillis = event.getPostDelaySeconds() * 1000L; // should be 180 days
+        long nowMillis = Instant.now().toEpochMilli();
+
+        LOG.info("Starting backfill with eventConfigId={}, delayMillis={}, nowMillis={}",
+                event.getEventConfigurationId(), delayMillis, nowMillis);
+
+        List<Candidate> candidates = helper.findNotCompletedOrNoQueuedEvent(studyDto.getId(), event.getEventConfigurationId());
+        LOG.info("Found {} participants that are eligible for COMPLETED or does not have a queued event for it yet", candidates.size());
+
+        List<Candidate> eligible = new ArrayList<>();
+        List<Candidate> needQueuing = new ArrayList<>();
+        for (var candidate : candidates) {
+            long millisWhenCompletedShouldHappen = candidate.firstEnrolledAtMillis + delayMillis;
+            if (millisWhenCompletedShouldHappen <= nowMillis) {
+                eligible.add(candidate);
+            } else {
+                needQueuing.add(candidate);
+            }
+        }
+
+        LOG.info("Number of eligible participants: {}", eligible.size());
+        LOG.info("Number of participants that need a queued event: {}", needQueuing.size());
+        System.out.print("Press enter to continue...");
+        System.out.flush();
+        new Scanner(System.in).nextLine();
+
+        LOG.info("Queuing future events now...");
+        var queueDao = handle.attach(QueuedEventDao.class);
+        for (var candidate : needQueuing) {
+            long millisWhenCompletedShouldHappen = candidate.firstEnrolledAtMillis + delayMillis;
+            long millisLeft = millisWhenCompletedShouldHappen - nowMillis;
+            if (millisLeft < 0) {
+                throw new DDPException("Something is wrong! Calculated future delay is less than zero!");
+            }
+            int delaySecondsLeft = (int) (millisLeft / 1000);
+            long queuedId = queueDao.addToQueue(event.getEventConfigurationId(),
+                    candidate.userId, candidate.userId, delaySecondsLeft);
+            LOG.info("Queued status change event for participant {} with queued_event_id={} after delay_seconds={}",
+                    candidate.userId, queuedId, delaySecondsLeft);
+        }
+
+        LOG.info("Handling eligible participants now...");
+        int batchSize = 100;
+        int bufferSecs = 10 * 60; // 10 mins
+        int delaySecsToUse = 0;
+        // Since there's likely a large number of these, we partition into batches and make each batch 10 mins apart.
```

Comment:
There's roughly 4500+ that need this, so roughly 45 batches. So it'll take 450 mins to run through, roughly 7 hrs. Is this okay?

Reply:
Bumping this up to 5 mins apart.